internal/tlscert/tls_cert_validator.go

```diff
@@ -21,16 +28,36 @@ type TLSCertValidationResult struct {
 	Validity                    time.Time
 }
 
-func (tcv *TLSCertValidator) ValidateCertificate(certPEM []byte, keyPEM []byte) TLSCertValidationResult {
+func New(client client.Client) *TLSCertValidator {
+	return &TLSCertValidator{
+		client: client,
+	}
+}
+
+func (tcv *TLSCertValidator) ResolveAndValidateCertificate(ctx context.Context, certPEM *telemetryv1alpha1.ValueType, keyPEM *telemetryv1alpha1.ValueType) TLSCertValidationResult {
```

Comment:
IMHO, ValidateCertificate would be still a good name. ResolveAndValidate seems to be too detailed.

Reply:
reverted